bitbots_path_planning/bitbots_path_planning/planner.py

```diff
@@ -0,0 +1,76 @@
+from typing import Optional
+
+import numpy as np
+import pyastar2d
+import tf2_geometry_msgs   # Is there for tf to recognize geomety_msgs even if it is not used explicitly
+import tf2_ros as tf2
+from bitbots_path_planning.map import Map
+from geometry_msgs.msg import PoseStamped
+from nav_msgs.msg import Path
+from rclpy.node import Node
+from rclpy.time import Time
+from std_msgs.msg import Header
+
+
+class Planner:
+    def __init__(self, node: Node, buffer: tf2.Buffer, map: Map) -> None:
+        self.node = node
+        self.buffer = buffer
+        self.map = map
+        self.goal = None
+        self.path = None
+
+    def set_goal(self, pose: PoseStamped) -> None:
+        pose.header.stamp = Time().to_msg()
+        self.goal = pose
+
+    def cancel(self) -> None:
+        self.goal = None
+
+    def active(self) -> bool:
+        return self.goal is not None
+
+    def step(self) -> Path:
+        navigation_grid = self.map.get_map()
+
+        my_pose = self.buffer.lookup_transform(
+            self.map.frame,
+            self.node.get_parameter("base_footprint_frame").value,
+            Time()).transform
+        my_position = my_pose.translation
+
+        # Transform header to map frame if needed
+        if self.goal.header.frame_id != self.map.frame:
+            self.goal = self.buffer.transform(self.goal, self.map.frame)
+
+        goal = self.goal
```

Comment:
Why do you use the local variable now?

Reply:
Because the goal might get updated, and we don't want this happening during the planning.